dsnp-wallet/dsnp-wallet/Source/ViewControllers/ProfileViewController.swift

```diff
@@ -5,10 +5,366 @@
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide), name:UIResponder.keyboardWillHideNotification, object: nil)
+        
+        let tap: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        view.addGestureRecognizer(tap)
+    }
+    
+    @objc private func keyboardWillShow(notification:NSNotification) {
+        guard let userInfo = notification.userInfo else { return }
+        guard var keyboardFrame: CGRect = (userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else { return }
+
+        keyboardFrame = view.convert(keyboardFrame, from: nil)
+        let keyboardHeight = keyboardFrame.height
+
+        //Move Save Btn
+        let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as! Double
+        let curve = notification.userInfo![UIResponder.keyboardAnimationCurveUserInfoKey] as! UInt
+        UIView.animate(
+            withDuration: duration,
+            delay: 0.0,
+            options: UIView.AnimationOptions.init(rawValue: curve),
+            animations: {
+                let height = self.view.safeAreaInsets.bottom - keyboardHeight
+                self.scrollViewBottomConstraint.constant = height
+                self.saveViewBottomConstraint.constant = height - CGFloat(self.saveBtnBottomOffset)
+                self.view.layoutIfNeeded()
+            }
+        )
+    }
+    
+    @objc private func keyboardWillHide(notification:NSNotification){
+        let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as! Double
+        let curve = notification.userInfo![UIResponder.keyboardAnimationCurveUserInfoKey] as! UInt
+        
+        UIView.animate(
+            withDuration: duration,
+            delay: 0.0,
+            options: UIView.AnimationOptions.init(rawValue: curve),
+            animations: {
+                self.scrollViewBottomConstraint.constant = 0
+                self.saveViewBottomConstraint.constant = CGFloat(-1 * self.saveBtnBottomOffset)
+                self.view.layoutIfNeeded()
+            }
+        )
+    }
+    
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension ProfileViewController: UITextFieldDelegate {
+    func textFieldDidChangeSelection(_ textField: UITextField) {
+        saveBtn(enabled: isEdited())
+    }
+}
+
+extension ProfileViewController: ProfileHeaderDelegate {
+    func tappedAvatar() {
+        presentActionSheet()
+    }
+}
+
+extension ProfileViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    private func presentActionSheet() {
+        self.view.endEditing(true)
+        
+        let actionSheetController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
```

Comment:
This alert/image handler is a good candidate to make as a standalone class.

Reply:
Or perhaps the alert can be presented here but the logic to call upon ImagePickerController can be implemented elsewhere as a standalone object. If you want to discuss this, let me know.